Add support for enums in server veriable object

siavashs opened the pull request:
Add support for enums in server variable object.
Variables with an `enum` are rendered as `select` with the `default` option set to `selected`.
Variables without an `enum` are rendered as `input` (like before).
The examples are updated to show the above cases.

Reference: https://swagger.io/specification/#server-variable-object

siavashs commented:
@mrin9 I confirm:
![image](https://user-images.githubusercontent.com/113189/85896645-28f36180-b7f9-11ea-9990-cb1a3a15ab38.png)

siavashs commented:
BTW, I think we also need an example with a variable which only has the `default` and no `enum`, that should create a text `input` instead.
I tested that case with my own `swagger.json` copy but I can also add one in the `server-vars.html`.

mrin9 commented:
> BTW, I think we also need an example with a variable which only has the `default` and no `enum`, that should create a text `input` instead.

sure go ahead, at the moment, we don't have any automated-tests, the examples serves us our test cases, so if some thing is missing then please update the example, or if you add any new functionality, you may add new example/tests to showcase/verify the functionality

siavashs commented:
@mrin9 done, I updated one of the existing examples to show the cases with/without `enum`s which are rendered as `select` or `input`.
I also added the logic to set the `default` `option` to `selected`.
